src/back/spv/writer.rs

```diff
@@ -439,6 +441,44 @@ impl Writer {
             results: Vec::new(),
         };
 
+        for (handle, expression) in ir_function.expressions.iter() {
+            match expression {
+                &crate::Expression::Compose { ty, .. } => {
```

Comment:
As discussed on the Matrix, this code is overly generic.
The problem we need to solve is about dynamic indexing of arrays that are given by values (i.e. expressions or constants).

Reply:
I think I have narrowed it down at this point. Thank you for identifying this so quickly!